compiler/ir/backend.py/src/org/jetbrains/kotlin/ir/backend/py/transformers/irToPy/IrModuleToPyTransformer.kt

```diff
@@ -84,20 +101,75 @@ class IrModuleToPyTransformer(
             localNames = LocalNameGenerator(NameScope.EmptyScope)
         )
 
+        val callToMain = generateCallToMain(modules)
+        val imports = generateImports()
+
         val moduleBody = generateModuleBody(modules, rootContext)
         val program = Module(
             body = moduleBody,
             type_ignores = emptyList(),
-        )
+        ).let {
+            it.copy(body = imports + it.body + callToMain)
```

Comment:
We can actually import inside `callToMain`. Would be a bit more compact both in our code and in generated code. Of course it won't be clean code, but I guess we aren't trying to generate clean code for now.

But I'm not insisting.